pkg/worker/k8srunner.go

```diff
@@ -379,27 +401,76 @@ func (r k8sStepRunner) readLogs(ctx context.Context, podName string, opts *v1.Po
 	return scanner.Err()
 }
 
-func (r k8sStepRunner) stopPodNow(ctx context.Context, stepName, podName string) {
+func (r k8sStepRunner) stopPodNow(ctx context.Context) {
 	gracePeriod := int64(0) // 0=immediately
-	err := r.pods.Delete(ctx, podName, metav1.DeleteOptions{
+	err := r.pods.Delete(ctx, r.target.name, metav1.DeleteOptions{
 		GracePeriodSeconds: &gracePeriod,
 	})
 	if err != nil {
 		log.Warn().
 			WithError(err).
-			WithString("step", stepName).
-			WithString("pod", podName).
+			WithString("step", r.step.Name).
+			WithString("pod", r.target.name).
 			Message("Failed to delete pod.")
 	} else {
 		log.Debug().
-			WithString("step", stepName).
-			WithString("pod", podName).
+			WithString("step", r.step.Name).
+			WithString("pod", r.target.name).
 			Message("Deleted pod.")
 	}
 }
 
-func (r k8sStepRunner) continueInitContainer(podName string) error {
-	exec, err := execInPodPipeStdout(r.RestConfig, r.Config.K8s.Namespace, podName, "init", podInitContinueArgs)
+func (r k8sStepRunner) transferDataToPod(ctx context.Context) error {
+	var logFunc = func(ev logger.Event) logger.Event {
+		return ev.
+			WithString("step", r.step.Name).
+			WithString("pod", r.target.name)
+	}
+
+	log.Debug().WithFunc(logFunc).Message("Transferring repo to init container.")
+	if err := r.copyDirToPod(ctx, commonRepoVolumeMount.MountPath); err != nil {
+		return fmt.Errorf("transfer repo: %w", err)
+	}
+	log.Debug().WithFunc(logFunc).Message("Transferred repo to init container.")
+
+	if step, ok := r.step.Type.(wharfyml.StepDocker); ok {
+		{
+			log.Debug().WithFunc(logFunc).Message("Transferring modified Dockerfile to init container.")
+			dockerfilePath := filepath.Join(r.CurrentDir, step.File)
+			if !validateNoParentDirAccess(dockerfilePath) {
+				return fmt.Errorf("transfer modified dockerfile: %w: %q", errParentDirAccessNotAllowed, dockerfilePath)
+			}
+			if _, err := os.Stat(dockerfilePath); err != nil {
+				return fmt.Errorf("transfer modified dockerfile: %w", err)
+			}
+			if err := r.copyDockerfileToPod(ctx, step); err != nil {
+				return fmt.Errorf("transfer modified dockerfile: %w", err)
+			}
+			log.Debug().WithFunc(logFunc).Message("Transferred modified Dockerfile to init container.")
+		}
+
+		{
+			log.Debug().WithFunc(logFunc).Message("Transferring cert file to init container.")
+			certFile, err := os.Open("/mnt/cert/root.crt")
+			if err != nil {
+				return fmt.Errorf("transfer cert: %w", err)
+			}
+			destPath := filepath.Join(commonRepoVolumeMount.MountPath, step.Context, "root.crt")
+			if !validateNoParentDirAccess(destPath) {
+				return fmt.Errorf("transfer cert: %w: %q", errParentDirAccessNotAllowed, destPath)
+			}
+			args := []string{"tee", destPath}
+			if err := r.copyToPodStdin(ctx, certFile, args); err != nil {
+				return fmt.Errorf("transfer cert: %w", err)
+			}
```

Comment:
The cert should live on the step pod's disk; not on the worker's. It should not be transferred. This to make it a dependency on the installation of the cluster, and not on your local environment.

To transfer the cert, you would do the equivalent of:

```sh
kubectl exec --namespace my-namespace my-pod -- cp -v /mnt/cert/root.crt /mnt/repo/root.crt